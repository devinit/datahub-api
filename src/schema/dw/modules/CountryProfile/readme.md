## Country profiles schema and resolvers


The data from DW should match the below interfaces


