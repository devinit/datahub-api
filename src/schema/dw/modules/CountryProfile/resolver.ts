import {
    IContext
} from '../../../../schema';

export default {
    Query: {
        async governmentFinance(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.resources.getGovernmentFinance(args);
        },
        async internationalResources(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.resources.getInternationalResources(args);
        },
        async singleResource(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.resources.getSingleResource(args);
        },
        async overViewTab(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.tabs.getOverViewTab(args);
        },
        async populationTab(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.tabs.getPopulationTab(args);
        },
        async povertyTab(_root, args, ctx: IContext) {
            return ctx.dw.countryProfile.tabs.getPovertyTab(args);
        }
    }
};
