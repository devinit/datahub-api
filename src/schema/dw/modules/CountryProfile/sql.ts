export default {
    population: 'SELECT value FROM fact.population_total WHERE di_id = ${id} AND year = ${startYear}',
    // tslint:disable-next-line:max-line-length
    domesticPublicResources: 'SELECT value FROM data_series.domestic WHERE di_id =${id} AND budget_type = \'actual\' AND year = ${startYear} AND l1 = \'total-revenue-and-grants\' AND l2 = \'revenue\' AND l3 is NULL',
    // tslint:disable-next-line:max-line-length
    internationalResources: 'SELECT value FROM data_series.intl_flows_recipients WHERE di_id = ${id} AND year = ${startYear} AND direction =\'in\'',
    governmentSpendPerPerson: 'SELECT value FROM data_series.govt_spend_pc WHERE di_id = ${id} AND year = ${startYear}',
    poorestPeople: 'SELECT value FROM data_series.poorest_20_percent WHERE di_id = ${id} AND year = ${startYear}',
    // tslint:disable-next-line:max-line-length
    averageIncomerPerPerson: 'SELECT value, year, di_id FROM fact.gni_pc_usd_2015 WHERE di_id = ${id} AND year >= ${startYear} AND year <= ${endYear}',
    incomeDistTrend: 'SELECT * FROM fact.income_share_by_quintile WHERE di_id = ${id} AND year >= ${startYear}'
};
