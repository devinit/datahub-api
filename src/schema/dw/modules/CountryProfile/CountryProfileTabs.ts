import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import sql from './sql';
import * as shortid from 'shortid';
import * as R from 'ramda';
import {getIndicatorData, IGetIndicatorArgs, isDonor, indicatorDataProcessingNamed, DONOR,
        IRAWPopulationAgeBand, normalizeKeyName, IRAW, IRAWQuintile, RECIPIENT, formatNumbers,
        IRAWPopulationGroup} from '../utils';

interface IOverViewTabRecipients {
    countryType: string;
    poorestPeople: string;
    population: string;
    domesticResources: string;
    governmentSpendPerPerson: string;
    internationalResources: string;
}
interface IOverViewTabDonors {
    countryType: string;
    governmentSpendPerPerson: string;
    averageIncomerPerPerson: DH.IIndicatorData[];
    incomeDistTrend: DH.IQuintile[];
}
const red = '#e8443a';
const grey = '#847e84';
export default class CountryProfileTabs {
    private db: IDatabase<IExtensions> & IExtensions;
    private defaultArgs;

    constructor(db: any) {
        this.db = db;
        this.defaultArgs = {db: this.db, conceptType: 'country-profile'};
    }

    public async getOverViewTab({id}): Promise<DH.IOverViewTab> {
        try {
            const isDonorCountry =  await isDonor(id);
            if (isDonorCountry) {
                const donorsTabData = await this.getOverViewTabDonors(id);
                return {
                    ...donorsTabData,
                    poorestPeople: null,
                    population: null, domesticResources: null, internationalResources: null};
            }
            const recipientTabData = await this.getOverViewTabRecipients(id);
            return {
                ...recipientTabData,
                averageIncomerPerPerson: null, incomeDistTrend: null, countryType: RECIPIENT};
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
    public async getPopulationTab({id}): Promise<DH.IPopulationTab> {
       try {
           const [population] = await this.getIndicatorsGeneric(id, [sql.population]);
           const populationDistribution = await this.getPopulationDistribution(id);
           const populationPerAgeBand = await this.getPopulationPerAgeBand(id);
           return {
            population,
            populationDistribution,
            populationPerAgeBand
          };
       } catch (error) {
           console.error(error);
           throw error;
       }
    }
    public async getPovertyTab({id}): Promise<DH.IPovertyTab> {
        try {
            const isDonorCountry =  await isDonor(id);
            if (isDonorCountry) return {poverty190Trend: null, depthOfExtremePoverty: null, incomeDistTrend: null};
            const poverty190Trend = await this.getPoverty190Trend(id);
            const [depthOfExtremePoverty] = await this.getIndicatorsGeneric(id, [sql.depthOfExtremePoverty], false);
            const incomeDistTrend = await this.getIncomeDistTrend(id);
            return {
                poverty190Trend,
                depthOfExtremePoverty,
                incomeDistTrend
            };
        } catch (error) {
            console.error(error);
            throw error;
        }
    }

    public async getOverViewTabRecipients(countryId: string): Promise<IOverViewTabRecipients> {
        try {
            const [internationalResources, domesticResources, population, poorestPeople, governmentSpendPerPerson]
            = await this.getIndicatorsGeneric(countryId,
                [sql.internationalResources, sql.domesticRevenue, sql.population,
                sql.poorestPeople, sql.governmentSpendPerPerson]);
            return {
                countryType: RECIPIENT,
                internationalResources,
                domesticResources,
                population,
                poorestPeople,
                governmentSpendPerPerson
            };
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
    public async getOverViewTabDonors(countryId: string): Promise<IOverViewTabDonors> {
        try {
            // tslint:disable-next-line:max-line-length
            const [governmentSpendPerPerson] = await this.getIndicatorsGeneric(countryId, [sql.governmentSpendPerPerson]);
            const averageIncomerPerPerson = await this.getAverageIncomerPerPerson(countryId);
            const incomeDistTrend = await this.getIncomeDistTrend(countryId);
            return {
                countryType: DONOR,
                governmentSpendPerPerson,
                averageIncomerPerPerson,
                incomeDistTrend
            };
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
    private async getIndicatorsGeneric(id: string, sqlList: string[], format: boolean = true)
        : Promise<string[]>  {
        try {
            const indicatorArgs: IGetIndicatorArgs[] =
                sqlList.map(query => ({...this.defaultArgs, query, id}));
            const indicatorRaw: IRAW[][] = await Promise.all(indicatorArgs.map(args => getIndicatorData<IRAW>(args)));
            return indicatorRaw.map(data => format ? formatNumbers(data[0].value, 1) : (data[0].value));
        } catch (error) {
            throw error;
        }
    }

    private async getAverageIncomerPerPerson(id): Promise<DH.IIndicatorData[]> {
         try {
             const indicatorArgs: IGetIndicatorArgs = {
             ...this.defaultArgs,
             query: sql.averageIncomerPerPerson,
             id
          };
             const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
             return indicatorDataProcessingNamed(data);
         } catch (error) {
             throw error;
         }
    }
    private async getIncomeDistTrend(id): Promise<DH.IQuintile[]> {
        try {
            const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultArgs,
            query: sql.incomeDistTrend,
            id
        };
            const data: IRAWQuintile[] = await getIndicatorData<IRAWQuintile>(indicatorArgs);
            const quintileData = R.last(data.filter(obj => obj.value_2nd_quintile !== null)) as IRAWQuintile;
            return R.keys(quintileData)
                .map(key => {
                    const color = key === 'value_bottom_20pc' ? red : grey;
                    return {quintileName: key, color, value: Number(data[0][key]), uid: shortid.generate()};
                });
        } catch (error) {
           throw error;
        }
    }
    private async getPopulationDistribution(id): Promise<DH.IPopulationDistribution[]> {
        try {
            const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultArgs,
            query: sql.populationDistribution,
            id
        };
            const data: IRAWPopulationGroup[] = await getIndicatorData<IRAWPopulationGroup>(indicatorArgs);
            return data.reduce((acc: DH.IPopulationDistribution[], row) => {
            const rural = {group: 'rural', value: Number(row.value_rural), year: Number(row.year) };
            const urban = {group: 'urban', value: Number(row.value_urban),  year: Number(row.year) };
            return [...acc, rural, urban];
        }, []);
       } catch (error) {
           throw error;
       }
    }
    private async getPopulationPerAgeBand(id): Promise<DH.IPopulationPerAgeBand[]> {
        try {
            const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultArgs,
            query: sql.populationPerAgeBand,
            id
        };
            const data: IRAWPopulationAgeBand[] = await getIndicatorData<IRAWPopulationAgeBand>(indicatorArgs);
            const groupColumns: string[] = R.keys(data[0]).filter(key => key !== 'year' && key !== 'di_id');
            return data.reduce((acc: DH.IPopulationPerAgeBand[], row) => {
            const groups: DH.IPopulationPerAgeBand[] = groupColumns
                .map(key => ({band: normalizeKeyName(key), value: Number(row[key]),
                    year: Number(row.year), uid: shortid.generate() }));
            return [...acc, ...groups];
          }, []);
        } catch (error) {
            throw error;
        }
    }
    private async getPoverty190Trend(id): Promise<DH.IIndicatorData[]> {
        try {
            const indicatorArgs: IGetIndicatorArgs = {
            query: sql.poverty190Trend,
            id,
            ...this.defaultArgs
        };
            const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
            return indicatorDataProcessingNamed(data);
        } catch (error) {
           throw error;
        }
    }
}
