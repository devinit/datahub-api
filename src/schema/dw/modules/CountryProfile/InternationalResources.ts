import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import {formatNumbers} from '../../../../utils';
import sql from './sql';
import * as R from 'ramda';
import {getIndicatorData, RECIPIENT, DONOR, IGetIndicatorArgs, isDonor, IRAW} from '../utils';

export default class CountryProfileTabs {
    private db: IDatabase<IExtensions> & IExtensions;
    private defaultDonorArgs;
    private defaultRecipientArgs;

    constructor(db: any) {
        this.db = db;
        this.defaultDonorArgs = {db: this.db, theme: DONOR};
        this.defaultRecipientArgs = {db: this.db, theme: RECIPIENT};
    }
    public async getOverViewTab({id}): Promise<DH.OverViewTab> {
        const IsDonor =  await isDonor(id);
        if (IsDonor) return this.getOverViewTabDonors(id);
        return this.getOverViewTabRecipients(id);
    }
}
