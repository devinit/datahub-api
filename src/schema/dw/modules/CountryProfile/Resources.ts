 // tslint:disable-next-line:no-reference
 /// <reference path="../../../../types/dh.d.ts" />
import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import {formatNumbers} from '../../../../utils';
import sql from './sql';
import * as R from 'ramda';
import {getIndicatorData, RECIPIENT, DONOR, IGetIndicatorArgs, isDonor, IRAW} from '../utils';

interface IflowTypes {
    inflows: DH.IFlow[];
    outflows: DH.IFlow[];
}
interface ISingleResourceArgs {
    resourceId: string;
    countryId: string;
    direction: string;
    groupById: string;
}
export default class Resources {
    private db: IDatabase<IExtensions> & IExtensions;
    private defaultDonorArgs;
    private defaultRecipientArgs;

    constructor(db: any) {
        this.db = db;
        this.defaultDonorArgs = {db: this.db, theme: DONOR};
        this.defaultRecipientArgs = {db: this.db, theme: RECIPIENT};
    }
    public async getInternationalResources({id}): Promise<any> {
        const isDonorCountry =  await isDonor(id);
        const netODAOfGNIIn = isDonorCountry ? null : await this.getNetODAOfGNIIn(id);
        const netODAOfGNIOut = isDonorCountry ? await this.getNetODAOfGNIOut(id) : null;
        const GNI = this.getGNI(id);
        const {outflows, inflows}  = await this.getFlows(id);
        const resourcesOverTime = await this.getResourcesOverTime(id);
        const mixOfResources = await this.getMixOfResources(id);
        return {
            GNI,
            netODAOfGNIIn,
            netODAOfGNIOut,
            resourcesOverTime,
            mixOfResources,
            inflows,
            outflows
        };
    }
    // internation resource
    public async getSingleResource({resourceId, countryId, direction}: ISingleResourceArgs):
        Promise<DH.ISingleResourceData> {
    }
    // TODO: check if entity hasDomestic data from entity file before making any request
    public async getGovernmentFinance(id: string): Promise<DH.IGovernmentFinance>{
        
    }
    private async getGNI(id: string): Promise<string> {}
    private async getNetODAOfGNIIn(id: string): Promise<number> {}
    private async getNetODAOfGNIOut(id: string): Promise<number> {}
    private async getFlows(id: string): Promise<IflowTypes> {}
    private async getResourcesOverTime(id: string): Promise<DH.IResourceData[]> {}
    private async getMixOfResources(id: string): Promise<DH.IIndicatorData[]> {}
}
