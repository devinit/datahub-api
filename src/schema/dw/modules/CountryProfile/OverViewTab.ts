 // tslint:disable-next-line:no-reference
 /// <reference path="../../../../types/dh.d.ts" />
import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import {formatNumbers} from '../../../../utils';
import {getEntityByIdAsync, IEntity} from '../../../cms/modules/global';
import sql from './sql';
import * as R from 'ramda';
import {getIndicatorData,
        RECIPIENT,
        DONOR,
        IGetIndicatorArgs,
        IRAW, IRAWQuintile, indicatorDataProcessingSimple, getTotal} from '../utils';

export default class OverViewTab {
    private db: IDatabase<IExtensions> & IExtensions;
    private defaultDonorArgs;
    private defaultRecipientArgs;

    constructor(db: any) {
        this.db = db;
        this.defaultDonorArgs = {db: this.db, theme: DONOR};
        this.defaultRecipientArgs = {db: this.db, theme: RECIPIENT};
    }
    public async getOverViewTab(opts): Promise<DH.OverViewTab> {
        const countryId = opts.id;
        const {donorRecipientType}: IEntity = await getEntityByIdAsync(countryId);
        if (donorRecipientType === DONOR) return this.getOverViewTabDonors(countryId);
        return this.getOverViewTabRecipients(countryId);
    }

    public async getOverViewTabRecipients(countryId: string): Promise<DH.IOverViewTabRecipients> {
        const internationalResources  = await this.getInternationalResources(countryId);
        const domesticPublicResources = await this.getDomesticPublicResources(countryId);
        const population = await this.getPopulation(countryId);
        const poorestPeople = await this.getPoorestPeople(countryId);
        const governmentSpendPerPerson = await this.getGovernmentSpendPerPerson(countryId, RECIPIENT);
        return {
            internationalResources,
            domesticPublicResources,
            population,
            poorestPeople,
            governmentSpendPerPerson
        };
    }
    public async getOverViewTabDonors(countryId: string): Promise<DH.IOverViewTabDonors> {
        const governmentSpendPerPerson = await this.getGovernmentSpendPerPerson(countryId, DONOR);
        const averageIncomerPerPerson = await this.getAverageIncomerPerPerson(countryId);
        const incomeDistTrend = await this.getIncomeDistTrend(countryId);
        return {
            governmentSpendPerPerson,
            averageIncomerPerPerson,
            incomeDistTrend
        };
    }
    private async getInternationalResources(id): Promise<string> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultRecipientArgs,
            table: 'data_series.intl_flows_recipients',
            query: sql.internationalResources,
            id,
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
        const totalResources: number = getTotal(data);
        return formatNumbers(totalResources, 1);
    }
    private async getDomesticPublicResources(id): Promise<string> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultRecipientArgs,
            table: 'data_series.domestic',
            query: sql.domesticPublicResources,
            id
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
        const domesticRevenue: number = Number(data[0].value);
        return formatNumbers(domesticRevenue, 1);
    }
    private async getPopulation(id): Promise<string> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultRecipientArgs,
            table: 'fact.population_total',
            query: sql.population,
            id
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
        const totalPopulation: number = Number(data[0].value);
        return formatNumbers(totalPopulation, 0);
    }
    private async getPoorestPeople(id): Promise<string> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultRecipientArgs,
            table: 'data_series.poorest_20_percent',
            query: sql.poorestPeople,
            id
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
        const poorestPeople: number = Number(data[0].value);
        return formatNumbers(poorestPeople, 1);
    }
    private async getGovernmentSpendPerPerson(id, theme): Promise<string> {
        const indicatorArgs: IGetIndicatorArgs = {
            table: 'data_series.govt_spend_pc',
            query: sql.governmentSpendPerPerson,
            id,
            db: this.db,
            theme
        };
        const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
        const poorestPeople: number = Number(data[0].value);
        return poorestPeople.toFixed(0);
    }
    private async getAverageIncomerPerPerson(id): Promise<DH.IIndicatorData[]> {
         const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultDonorArgs,
            table: 'fact.gni_pc_usd_2015',
            query: sql.averageIncomerPerPerson,
            id
        };
         const data: IRAW[] = await getIndicatorData<IRAW>(indicatorArgs);
         return indicatorDataProcessingSimple<DH.IIndicatorData>(data);
    }
    private async getIncomeDistTrend(id): Promise<DH.IQuintile[]> {
        const indicatorArgs: IGetIndicatorArgs = {
            ...this.defaultDonorArgs,
            table: 'fact.income_share_by_quintile',
            query: sql.incomeDistTrend,
            id
        };
        const data: IRAWQuintile[] = await getIndicatorData<IRAWQuintile>(indicatorArgs);
        return R.keys(data[0]).map(key => ({quintileName: key, value: Number(data[0][key])}));
    }
}
