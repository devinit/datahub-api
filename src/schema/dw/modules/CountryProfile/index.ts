import {IDatabase} from 'pg-promise';
import {IExtensions} from '../../db';
import OverViewTab from './OverviewTab';

export default class CountryProfile {
    public overViewTab: OverViewTab;
    constructor(db: IDatabase<IExtensions> & IExtensions) {
        this.overViewTab = new OverViewTab(db);
    }
}
