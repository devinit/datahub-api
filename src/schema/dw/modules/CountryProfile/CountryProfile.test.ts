import * as prettyFormat from 'pretty-format';
// import {uidPatchForTabData} from '../utils/test-utils';
import CountryProfile from '.';
import * as R from 'ramda';
import db from '../../db';

describe('country profile DW module tests', () => {
    const countryProfile = new CountryProfile(db);
    const tab = countryProfile.tabs;
    const resources = countryProfile.resources;
    it('should return flow types and sub selections for both recipient & donors', async () => {
        const recipient = await resources.getFlows('recipient');
        const donor = await resources.getFlows('donor');
        expect(prettyFormat({recipient, donor})).toMatchSnapshot();
    }, 10000);
    it('should return overview tab data for uganda', async () => {
        const overViewTab = await tab.getOverViewTab({id: 'uganda'});
        expect(prettyFormat(overViewTab)).toMatchSnapshot();
    }, 10000);
    it('should return overview tab data for Austria', async () => {
        const overViewTab = await tab.getOverViewTab({id: 'austria'});
        expect(prettyFormat(overViewTab)).toMatchSnapshot();
    }, 10000);
    it('should return population tab data for Austria', async () => {
        const populationTab = await tab.getPopulationTab({id: 'austria'});
        expect(prettyFormat(populationTab)).toMatchSnapshot();
    }, 10000);
    it('should return population tab data for Uganda', async () => {
        const populationTab = await tab.getPopulationTab({id: 'uganda'});
        expect(prettyFormat(populationTab)).toMatchSnapshot();
    }, 10000);
    it('should return poverty tab data for Uganda', async () => {
        const povertyTab = await tab.getPovertyTab({id: 'uganda'});
        expect(prettyFormat(povertyTab)).toMatchSnapshot();
    }, 10000);
    it('should return international resources tab & charts data for Uganda', async () => {
        const internationalA = await resources.getInternationalResources({id: 'uganda'});
        const internationalB = await resources.getInternationalResources({id: 'somalia'});
        expect(prettyFormat({ uganda: internationalA, somalia: internationalB})).toMatchSnapshot();
    }, 30000);
    it('should return international resources tab & charts data for Austria', async () => {
        const international  = await resources.getInternationalResources({id: 'austria'});
        expect(prettyFormat(international)).toMatchSnapshot();
    }, 30000);
    it('should return government finance data for Uganda && somalia', async () => {
        const governmentA = await resources.getGovernmentFinance({id: 'uganda'});
        const governmentB = await resources.getGovernmentFinance({id: 'somalia'});
        expect(prettyFormat({somalia: governmentB, uganda: governmentA})).toMatchSnapshot();
    }, 10000);
    it('should return single resource FDI data for use in international resources chart', async () => {
        const singleResourceFDIout = await resources.getSingleResource(
            { resourceId: 'fdi-out', countryId: 'UG', groupById: 'from_di_id'});
        expect(prettyFormat(singleResourceFDIout)).toMatchSnapshot();
    }, 10000);
    it('should return single resource ODA data for use in international resources chart', async () => {
        const singleResourceODA = await resources.getSingleResource(
            { resourceId: 'oda-in', countryId: 'UG', groupById: 'sector'});
        expect(prettyFormat(singleResourceODA)).toMatchSnapshot();
    }, 10000);

    afterAll(() => {
       db.$config.pgp.end();
    });
});
