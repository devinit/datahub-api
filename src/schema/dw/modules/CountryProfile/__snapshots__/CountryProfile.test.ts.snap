// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`country profile DW module tests should return overview tab data for Austria 1`] = `
"Object {
  \\"averageIncomerPerPerson\\": Array [
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 47410,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 42526.39,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 43543.15,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 45059.11,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 42855.89,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 44915.97,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 42658.3,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 41207.23,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 40812.68,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 42523.58,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 40757.56,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 36577.17,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 34020.43,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 36852.26,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 39809.23,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 41512.09,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 37166.4,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 36648.38,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 38403.04,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 34302.92,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 30804.85,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 32903.33,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 32758.83,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 31285.07,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 30549.21,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 29209.85,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 32784.44,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 29483.17,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 24477.1,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 23584.09,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 28301.28,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 28074.77,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 27338.63,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 29138.65,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 30743.1,
      \\"year\\": null,
    },
    Object {
      \\"id\\": \\"AT\\",
      \\"value\\": 27360.55,
      \\"year\\": null,
    },
  ],
  \\"governmentSpendPerPerson\\": \\"24274\\",
  \\"incomeDistTrend\\": Array [
    Object {
      \\"quintileName\\": \\"di_id\\",
      \\"value\\": NaN,
    },
    Object {
      \\"quintileName\\": \\"year\\",
      \\"value\\": 2015,
    },
    Object {
      \\"quintileName\\": \\"value_bottom_20pc\\",
      \\"value\\": 0,
    },
    Object {
      \\"quintileName\\": \\"value_2nd_quintile\\",
      \\"value\\": 0,
    },
    Object {
      \\"quintileName\\": \\"value_3rd_quintile\\",
      \\"value\\": 0,
    },
    Object {
      \\"quintileName\\": \\"value_4th_quintile\\",
      \\"value\\": 0,
    },
    Object {
      \\"quintileName\\": \\"value_5th_quintile\\",
      \\"value\\": 0,
    },
  ],
}"
`;

exports[`country profile DW module tests should return overview tab data for uganda 1`] = `
"Object {
  \\"domesticPublicResources\\": \\"3.1bn\\",
  \\"governmentSpendPerPerson\\": \\"365\\",
  \\"internationalResources\\": \\"4.4bn\\",
  \\"poorestPeople\\": \\"19.6m\\",
  \\"population\\": \\"39m\\",
}"
`;
